src/scenes/Dashboard/ProgressReportsWidget/RichTextCell.tsx

```diff
@@ -0,0 +1,96 @@
+import { Box, Typography } from '@mui/material';
+import {
+  GridState,
+  GridRenderCellParams as RenderCellParams,
+  useGridSelector,
+} from '@mui/x-data-grid';
+import { extendSx, RichTextJson, StyleProps } from '~/common';
+import {
+  RenderFn,
+  RichTextRenderers,
+  RichTextView,
+  Text,
+} from '../../../components/RichText';
+import { ProgressReportsDataGridRowFragment as ProgressReport } from './progressReportsDataGridRow.graphql';
+
+type CellParams = RenderCellParams<ProgressReport, RichTextJson>;
+
+export const RichTextCell = ({
+  id,
+  value,
+  api,
+  sx,
+  className,
+}: Pick<CellParams, 'value' | 'id' | 'api'> & StyleProps) => {
+  const selectedRows = useGridSelector(
+    { current: api },
+    (state: GridState) => state.rowSelection
+  );
+  const isExpanded = selectedRows[0] === id;
+
+  if (!value) return null;
+
+  return (
+    <Box
+      sx={[
+        {
+          overflow: 'hidden',
+          textWrap: 'wrap',
+          display: isExpanded ? 'contents' : '-webkit-box',
+          WebkitLineClamp: '2',
+          WebkitBoxOrient: 'vertical',
+
+          // No trailing spacing on response
+          '& > *:last-child': { mb: 0 },
+        },
+        ...extendSx(sx),
+      ]}
+      className={className}
+    >
+      <RichTextView data={value} renderers={renderers} />
+    </Box>
+  );
+};
+
+const ParagraphBlock: RenderFn<{ text: string }> = ({ data }) => (
+  <Typography variant="body2" paragraph>
+    <Text data={data} />
+  </Typography>
+);
+
+const List: RenderFn<{ style: 'unordered' | 'ordered'; items: string[] }> = ({
+  data,
+}) => {
+  // eslint-disable-next-line react/jsx-no-useless-fragment
+  if (!data) return <></>;
```

Comment:
Why is List handling no data?  Can we handle this elsewhere?

Reply:
We could. But would need more work, in a different PR. Currently we don't own the component that calls these components.

Reply:
Actually no we couldn't. Because certain blocks could have no data but that doesn't mean the block shouldn't render. Like a `delimiter` just renders a `Divider`, no data needed for that.